server/manager/src/main/java/org/apache/accumulo/manager/BalanceManager.java

```diff
@@ -0,0 +1,331 @@
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import org.apache.accumulo.core.conf.DeprecatedPropertyUtil;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.core.dataImpl.KeyExtent;
+import org.apache.accumulo.core.manager.balancer.AssignmentParamsImpl;
+import org.apache.accumulo.core.manager.balancer.BalanceParamsImpl;
+import org.apache.accumulo.core.manager.balancer.TServerStatusImpl;
+import org.apache.accumulo.core.manager.balancer.TabletServerIdImpl;
+import org.apache.accumulo.core.manager.thrift.TableInfo;
+import org.apache.accumulo.core.manager.thrift.TabletServerStatus;
+import org.apache.accumulo.core.metadata.SystemTables;
+import org.apache.accumulo.core.metadata.TServerInstance;
+import org.apache.accumulo.core.metadata.schema.Ample;
+import org.apache.accumulo.core.metadata.schema.TabletMetadata;
+import org.apache.accumulo.core.metrics.MetricsProducer;
+import org.apache.accumulo.core.spi.balancer.BalancerEnvironment;
+import org.apache.accumulo.core.spi.balancer.TableLoadBalancer;
+import org.apache.accumulo.core.spi.balancer.TabletBalancer;
+import org.apache.accumulo.core.spi.balancer.data.TServerStatus;
+import org.apache.accumulo.core.spi.balancer.data.TabletMigration;
+import org.apache.accumulo.core.spi.balancer.data.TabletServerId;
+import org.apache.accumulo.manager.metrics.BalancerMetrics;
+import org.apache.accumulo.server.ServerContext;
+import org.apache.accumulo.server.manager.balancer.BalancerEnvironmentImpl;
+import org.apache.accumulo.server.manager.state.UnassignedTablet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+public class BalanceManager {
+
+  private static final Logger log = LoggerFactory.getLogger(BalanceManager.class);
+
+  private final Manager manager;
+  protected volatile TabletBalancer tabletBalancer;
+  private final BalancerEnvironment balancerEnvironment;
+  private final BalancerMetrics balancerMetrics = new BalancerMetrics();
+  private final Object balancedNotifier = new Object();
+
+  BalanceManager(Manager manager) {
+    this.manager = manager;
+    this.balancerEnvironment = new BalancerEnvironmentImpl(manager.getContext());
+    initializeBalancer();
+  }
+
+  private void initializeBalancer() {
+    var localTabletBalancer =
+        Property.createInstanceFromPropertyName(getContext().getConfiguration(),
+            Property.MANAGER_TABLET_BALANCER, TabletBalancer.class, new TableLoadBalancer());
+    localTabletBalancer.init(balancerEnvironment);
+    tabletBalancer = localTabletBalancer;
+  }
+
+  void propertyChanged(String property) {
+    String resolved = DeprecatedPropertyUtil.getReplacementName(property, (log, replacement) -> {});
```

Comment:
Is there still an old deprecated property in 4.0? If so, should we be looking for the deprecated property in `initializeBalancer`?

Reply:
This code is probably no longer needed.

Reply:
removed in 705d58b